test/src/util/pelz_json_parser_test_suite.c

```diff
@@ -0,0 +1,440 @@
+  freeCharBuf(&key_id);
+  freeCharBuf(&data);
+  cJSON_DeleteItemFromObject(json, "dec_data");
+  cJSON_AddItemToObject(json, "dec_data", cJSON_CreateString(dec_data));
+
+  //Test check of JSON request string length match
+  cJSON_DeleteItemFromObject(json, "key_id_len");
+  cJSON_AddItemToObject(json, "key_id_len", cJSON_CreateNumber(20));
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 1);
+  cJSON_DeleteItemFromObject(json, "key_id_len");
+  cJSON_AddItemToObject(json, "key_id_len", cJSON_CreateNumber(json_key_id_len));
+
+  cJSON_DeleteItemFromObject(json, "enc_data_len");
+  cJSON_AddItemToObject(json, "enc_data_len", cJSON_CreateNumber(50));
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 1);
+  cJSON_DeleteItemFromObject(json, "enc_data_len");
+  cJSON_AddItemToObject(json, "enc_data_len", cJSON_CreateNumber(enc_data_len));
+
+  //Clean-up JSON
+  cJSON_Delete(json);
+}
+
+void test_decrypt_parser(void)
+{
+  cJSON *json;
+  CharBuf key_id;
+  CharBuf data;
+
+  //Valid Test Values
+  char *json_key_id = "file:/test/key1.txt";
+  int json_key_id_len = 19;
+  char *enc_data = "QUJDREVGR0hJSktMTU5PUFFSU1RVVldYWVoxMjM0NTY=\n";
+  int enc_data_len = 45;
+  char *dec_data = "SwqqSZbNtN2SOfKGtE2jfklrcARSCZE9Tdl93pggkIsRkY3MrjevmQ==\n";
+  int dec_data_len = 57;
+
+  //Building of a standard valid JSON request
+  json = cJSON_CreateObject();
+  cJSON_AddItemToObject(json, "request_type", cJSON_CreateNumber(2));
+  cJSON_AddItemToObject(json, "key_id", cJSON_CreateString(json_key_id));
+  cJSON_AddItemToObject(json, "key_id_len", cJSON_CreateNumber(json_key_id_len));
+  cJSON_AddItemToObject(json, "enc_data", cJSON_CreateString(enc_data));
+  cJSON_AddItemToObject(json, "enc_data_len", cJSON_CreateNumber(enc_data_len));
+  cJSON_AddItemToObject(json, "dec_data", cJSON_CreateString(dec_data));
+  cJSON_AddItemToObject(json, "dec_data_len", cJSON_CreateNumber(dec_data_len));
+
+  //Test standard valid JSON request
+  CU_ASSERT(decrypt_parser(json, &key_id, &data) == 0);
+  freeCharBuf(&key_id);
+  freeCharBuf(&data);
+
+  //Test check of JSON request hasObject
+  cJSON_DeleteItemFromObject(json, "key_id");
+  CU_ASSERT(decrypt_parser(json, &key_id, &data) == 1);
+  cJSON_AddItemToObject(json, "key_id", cJSON_CreateString(json_key_id));
+
+  cJSON_DeleteItemFromObject(json, "key_id_len");
+  CU_ASSERT(decrypt_parser(json, &key_id, &data) == 1);
+  cJSON_AddItemToObject(json, "key_id_len", cJSON_CreateNumber(json_key_id_len));
+
+  cJSON_DeleteItemFromObject(json, "dec_data");
+  CU_ASSERT(decrypt_parser(json, &key_id, &data) == 1);
+  cJSON_AddItemToObject(json, "dec_data", cJSON_CreateString(dec_data));
+
+  cJSON_DeleteItemFromObject(json, "dec_data_len");
+  CU_ASSERT(decrypt_parser(json, &key_id, &data) == 1);
+  cJSON_AddItemToObject(json, "dec_data_len", cJSON_CreateNumber(dec_data_len));
+
+  cJSON_DeleteItemFromObject(json, "enc_data");
+  CU_ASSERT(decrypt_parser(json, &key_id, &data) == 0);
```

Comment:
This makes a little more sense to me than the tests in `encrypt_parser`, since here we're explicitly testing that `decrypt_parser` ignores these fields it doesn't use. I'm still not sure that needs to be tested though.